Mutagen.Bethesda.Analyzers.Engine/Config/ConfigDirectoryProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Mutagen.Bethesda.Environments.DI;
+using Noggog;
+using Noggog.IO;
+
+namespace Mutagen.Bethesda.Analyzers.Config;
+
+public class ConfigDirectoryProvider(
+    IDataDirectoryProvider dataDirectoryProvider,
```

Comment:
Yeah, primary constructors are a bit flawed. I like to use them anyway because I know about their problems but of course that can lead to problems. Let me know if you want to change it.